cmd/maestro/agent/cmd.go

```diff
@@ -47,53 +58,98 @@ func NewAgentCommand() *cobra.Command {
 	fs := cmd.PersistentFlags()
 	fs.SetNormalizeFunc(utilflag.WordSepNormalizeFunc)
 	fs.AddGoFlagSet(flag.CommandLine)
-	commonOptions.CommoOpts.AddFlags(fs)
-	addFlags(fs)
+	agentOptions.AddFlags(fs)
+
 	utilruntime.Must(features.SpokeMutableFeatureGate.Add(ocmfeature.DefaultSpokeWorkFeatureGates))
 	utilruntime.Must(features.SpokeMutableFeatureGate.Set(fmt.Sprintf("%s=true", ocmfeature.RawFeedbackJsonString)))
 
 	return cmd
 }
 
-func runAgent(cmd *cobra.Command, args []string) {
-	ctx, cancel := context.WithCancel(context.Background())
-
-	stopCh := make(chan os.Signal, 1)
-	signal.Notify(stopCh, syscall.SIGINT, syscall.SIGTERM)
-	go func() {
-		defer cancel()
-		<-stopCh
-	}()
+type AgentOptions struct {
+	CommonOptions  *options.AgentOptions
+	WorkOptions    *spoke.WorkloadAgentOptions
+	KubeConfigFile string
+	Namespace      string
+}
 
+func NewAgentOptions() *AgentOptions {
+	workOptions := spoke.NewWorkloadAgentOptions()
+	// use 1M as the default limit for state feedback
+	workOptions.MaxJSONRawLength = 1024 * 1024
```

Comment:
I remembered it was set in agent options. Do you need to reset it again?

Reply:
yes, but the default value is small for maestro case, we need increase it here